src/arguments.cpp

```diff
@@ -314,40 +301,108 @@ GIArgument Args::type_to_g_type(GITypeInfo &argument_type_info, Local<Value> js_
+                // that doesn't have an out argument for the array's length.
+                // i.e. it returns a null terminated array or a fixed length array.
+                if (g_type_info_is_zero_terminated(type)) {
+                    // if the array is null terminated we can use a standard
+                    // string length method to find the array length
+                    length = g_strv_length(static_cast<gchar **>(native_array));
+                } else {
+                    // otherwise, if there's no length (length == -1) and
+                    // the array wasn't null terminated, it must have a fixed
+                    // size.
+                    length = g_type_info_get_array_fixed_size(type);
+                    if (length == -1) {
+                        // otherwise, if the array wasn't fixed size then we've exhausted
+                        // all the ways we available to find the array length.
+                        stringstream message;
+                        message << "unable to determine array length for C array";
+                        throw UnsupportedGIType(message.str());
+                    }
                 }
-                return js_array;
-            } else {
-                // use array_length param once the layers above this
-                // pas it correctly.
-                // the length param comes from the native function call's
-                // out arguments but, currently, it doesn't get passed own
-                // here correctly.
-                throw UnsupportedGIType("Converting non null terminated arrays is not yet supported");
             }
             break;
+        case GI_ARRAY_TYPE_ARRAY:
+        case GI_ARRAY_TYPE_BYTE_ARRAY: {
+            GArray *g_array = static_cast<GArray *>(native_array);
+            native_array = g_array->data;
+            length = g_array->len;
+            element_size = g_array_get_element_size(g_array);
+        } break;
+        case GI_ARRAY_TYPE_PTR_ARRAY: {
+            GPtrArray *ptr_array = static_cast<GPtrArray *>(native_array);
+            native_array = ptr_array->pdata;
+            length = ptr_array->len;
+            element_size = sizeof(gpointer);
+            break;
+        }
         default:
             throw UnsupportedGIType("cannot convert native array type");
     }
 
-    stringstream message;
-    message << "Converting array of type '" << g_type_tag_to_string(param_tag);
-    message << "' is not supported";
-    throw UnsupportedGIType(message.str());
+    if (native_array == nullptr || length == 0) {
+        return Nan::New<Array>(); // an empty array
+    }
+
+    GIArgument element;
+    Local<Array> js_array = Nan::New<Array>();
+    for (int i = 0; i < length; i++) {
+        // void** pointer = static_cast<void**>(static_cast<ulong>(native_array) + i * element_size);
+        void **pointer = static_cast<void **>(native_array + i * element_size);
```

Comment:
Awesome if it works for you, but on my side, arithemics on pointers are byte based. E.g. if I add `(void*)0x2000 + (int)1`, I get `(void*)0x2008`.